src/components/LocationAutocomplete/LocationAutocompleteRenderer/LocationAutocompleteRenderer.js

```diff
@@ -0,0 +1,174 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import bem from 'bem';
+import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
+import { faMapMarkerAlt } from '@fortawesome/free-solid-svg-icons';
+import { Autosuggest, Alert, ListItem, MarkedText } from '../../../index';
+import useDebounce from '../../../hooks/useDebounce';
+import POWERED_BY_GOOGLE_ON_WHITE from '../../../images/powered_by_google_on_white.png';
+import styles from './LocationAutocompleteRenderer.scss';
+
+const { elem } = bem({
+    name: 'LocationAutocompleteRenderer',
+    classnames: styles,
+});
+
+const DEBOUNCE_DELAY = 350;
+const ACCEPTABLE_API_STATUSES = ['OK', 'NOT_FOUND', 'ZERO_RESULTS'];
+
+const LocationAutocompleteRenderer = props => {
+    const [inputValue, setInputValue] = React.useState('');
+    const [suggestionsList, setSuggestionsList] = React.useState(null);
+    const [isLoading, setIsLoading] = React.useState(false);
+
+    const {
+        onSelectionChange,
+        inputPlaceholder,
+        noSuggestionsPlaceholder,
+        country,
+        placeTypes,
+        showCountryInSuggestions,
+        onError,
+        ...rest
+    } = props;
+
+    // Suggestion functions
+    const resetSuggestionsList = () => setSuggestionsList(null);
+    const suggestionToString = suggestion => (suggestion ? suggestion.description : '');
+
+    const debouncedInputValue = useDebounce(inputValue, DEBOUNCE_DELAY);
+
+    React.useEffect(() => {
+        if (debouncedInputValue) {
+            const service = new window.google.maps.places.AutocompleteService();
+
+            service.getPlacePredictions(
+                {
+                    input: debouncedInputValue,
+                    types: placeTypes,
+                    componentRestrictions: { country },
+                },
+                (predictions, status) => {
+                    if (ACCEPTABLE_API_STATUSES.includes(status)) {
+                        setSuggestionsList(predictions);
+                    } else {
+                        // TODO: check desired behaviour with Carlo
+                        // currently the UI will look same as when no suggestions found
+                        resetSuggestionsList();
+                        if (onError) {
+                            onError(status);
+                        }
+                    }
+                    setIsLoading(false);
+                }
+            );
+        } else {
+            resetSuggestionsList();
+        }
+    }, [country, debouncedInputValue, onError, placeTypes]);
```

Comment:
Tell me please why we need `onError` in this array?

Reply:
because it is used within the effect function